File: InfixtoPostfix.c
#include<stdio.h>
#include<stdlib.h>
#include<string.h>
# define max 100
char stack[max];
int top=-1;
void push(char ch)
{
    if(top==max-1)
    {
        printf("Stack Overflow\n");
        return ;
    }
    top++;
    stack[top]=ch;
}
char pop()
{
    if(top==max-1)
    {
        printf("Stack Overflow\n");
        exit(0);
    }
    return stack[top--];
}
int is_operator(char ch)
{
    if(ch=='+'||ch=='-'||ch=='*'||ch=='/'||ch=='^')
        return 1;
    else
        return 0;
}
int precedence(char ch)
{
    switch(ch)
    {
        case '^':return 3;break;
        case '/':case '*':return 2;break;
        case '+':case '-':return 1;break;
        default :return 0;
    }
}
int DigAlpha(char ch)
{
    if(ch>='0'&&ch<='9')
        return 1;
    else if(ch>=65&&ch<=90)
        return 1;
    else if(ch>='a'&&ch<='z')
        return 1;
    else return 0;
}
void infixtopostfix(char infix[],char postfix[])
{
    int i=0,j=0;
    push('(');
    strcat(infix,")");
    char ch,item=infix[i];
    while(item!='\0')
    {
        if(item=='(')
            push(item);
        else if(DigAlpha(item)==1)
        {
            postfix[j]=item;
            j++;
        }
        else if(is_operator(item)==1)
        {
            ch=pop();
            while(is_operator(ch)==1&&precedence(ch)>=precedence(item))
            {
             postfix[j]=ch;
             j++;
             ch=pop();
            }
            push(ch);
            push(item);
        }
        else if(item==')')
        {
            ch=pop();
            while(ch!='(')
            {
                postfix[j]=ch;
                j++;
                ch=pop();
            }
        }
        i++;
        item=infix[i];
    }
    postfix[j]='\0';
}
int main()
{
    char infix[100],postfix[100];
    printf("Enter infix Expression: ");
    gets(infix);
    infixtopostfix(infix,postfix);
    printf("Postfix Expression: %s",postfix);
}

File: InfixtoPostfix.java
import java.util.*;
class node
{
    char data;
    node next;
    node(char data)
    {
        this.data=data;
    }
}
class Stack
{
    private node top;
    void push(char ele)
    {
        if(top==null)
        {
            top=new node(ele);
           // top.data=ele;
            top.next=null;
            return;
        }
        node curr=new node(ele);
        //curr.data=ele;
        curr.next=top;
        top=curr;
    }
    char pop()
    {
        if(top==null)
        {
            System.out.println("Stack Underflow,no Element to delete");
           // return;
        }
        //System.out.println("Delete Element = "+top.data);
        node ptr=top;
        top=top.next;
        return ptr.data;
    }
    char peek()
    {
        if(top==null)
        {
            System.out.println("Stack Underflow,no Element");
            //return;
        }
        //System.out.println("top Element = "+top.data);
        return top.data;
    }
    void display()
    {
        if(top==null)
        {
            System.out.println("Stack Underflow,no Element to display");
            return;
        }
        System.out.print("Elements are = ");
        node temp=top;
        while(temp!=null)
        {
            System.out.print(temp.data+" ");
            temp=temp.next;
        }
    }
}
class demo
{
    public static void main(String ar[])
    {
        Scanner sc=new Scanner(System.in);
        Stack s=new Stack();
        System.out.print("Enter Infix Expression: ");
        String post="",infix=sc.next();
        infix="("+infix+")";
        int no1,no2,no;
        for(int i=0;i<infix.length();i++)
        {
           
           char item,ch=infix.charAt(i);
           if(infix.charAt(i)>='a'&&infix.charAt(i)<='z')
           post+=infix.charAt(i);
           else if(infix.charAt(i)=='(')
           s.push('(');
           else if(ch=='+'||ch=='-'||ch=='*'||ch=='/'||ch=='^')
            {
                item=s.pop();
                while((item=='+'||item=='-'||item=='*'||item=='/'||item=='^')&&precdenece(item)>=precdenece(ch))
                {
                     post+=item;
                     item=s.pop();
                }
                //else
                    s.push(item);
                s.push(ch);
            } 
            else if(ch==')')
            {
                item=s.pop();
                while(item!='(')
                {
                    post+=item;
                    item=s.pop();
                }
            }           
          // s.display();
        }
        System.out.println("\n Postfix Expression = "+post);
    }
     static int precdenece(char ch)
		{
                  if(ch=='^')
					  return 3;
				  else if(ch=='*'||ch=='/')
					  return 2;
				  else if(ch=='+'||ch=='-')
					  return 1;
				  else
					  return 0;
		}
}